File: 07 - 10001st Prime/rust/main.rs
fn main() {
    let result = get_nth_prime(10001);
    println!("result: {}", result);
}

fn sieve(num: u32) -> Vec<u32> {
    let mut primes: Vec<bool> = Vec::new();
    for _ in 1..=num {
        primes.push(true);
    }
    let mut i: u32 = 2;
    while i.pow(2) <= num {
        if primes[i as usize] == true {
            for j in (i.pow(2)..num).step_by(i as usize) {
                primes[j as usize] = false;
            }
        }
        i += 1;
    }
    let mut result: Vec<u32> = Vec::new();
    for k in 2..num {
        if primes[k as usize] == true {
            result.push(k);
        }
    }
    result
}

fn get_nth_prime(nth: u32) -> u32 {
    let mut count: usize = 0;
    let mut size: u32 = 2;
    let mut limit: u32 = nth * size;
    let mut primes: Vec<u32> = Vec::new();
    while count < nth as usize {
        primes = sieve(limit);
        count = primes.len();
        size += 1;
        limit = nth * size;
    }
    primes[(nth-1) as usize]
}

File: 07 - 10001st Prime/cpp/main.cpp
#include <iostream>
#include <vector>
#include <cmath>

std::vector<uint> sieve(uint num)
{
    std::vector<bool> primes = {};
    for (int i = 0; i <= num; i++)
    {
        primes.push_back(true);
    }
    uint i = 2;
    while (pow(i, 2) <= num)
    {
        if (primes[i] == true)
        {
            for (int j = pow(i, 2); j < num; j += i)
            {
                primes[j] = false;
            }
        }
        i++;
    }
    std::vector<uint> result = {};
    for (int k = 2; k < num; k++)
    {
        if (primes[k] == true)
        {
            result.push_back(k);
        }
    }
    return result;
}

uint getNthPrime(uint nth)
{
    uint count = 0;
    uint size = 2;
    uint limit = nth * size;
    std::vector<uint> primes = {};
    while (count < nth)
    {
        primes = sieve(limit);
        count = primes.size();
        size++;
        limit = nth * size;
    }
    return primes[nth-1];
}

int main()
{
    uint result = getNthPrime(10001);
    std::cout << "result: " << result << std::endl;
}